llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp

```diff
@@ -1770,6 +1788,15 @@ class AArch64Operand : public MCParsedAsmOperand {
     Inst.addOperand(MCOperand::createReg(AArch64::Z0 + getReg() - Base));
   }
 
+  void addPPRorPNRRegOperands(MCInst &Inst, unsigned N) const {
+    assert(N == 1 && "Invalid number of operands!");
+    unsigned Reg = getReg();
+    // Normalise to PPR
+    if (Reg >= AArch64::PN0)
+      Reg = Reg - AArch64::PN0 + AArch64::P0;
+    Inst.addOperand(MCOperand::createReg(Reg));
+  }
+
```

Comment:
Can `addPNRasPPRRegOperands` be removed now? Or perhaps I should ask: is `PNRasPPR` still required, or can those instructions that use it also use `PPRorPNRRegOperand` ?

Reply:
`PPRasPNR` can be removed. Thanks for the idea.